e2e/stress/DeviceTelemetryTest/DeviceTelemetryTest.cs

```diff
@@ -0,0 +1,131 @@
+
+using Microsoft.Azure.Devices.Client;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class DeviceTelemetryTest
+    {
+        int _messageSize;
+        int _messageCount;
+        int _parallelCount;
+        TransportType _protocol;
+        ResultWriter _writer;
+        DeviceClient _client;
+        Stopwatch _wall = new Stopwatch();
+
+        public DeviceTelemetryTest(
+            TransportType protocol,
+            int messageCount,
+            int messageSize,
+            int parallelCount,
+            ResultWriter resultWriter)
+        {
+            _messageCount = messageCount;
+            _messageSize = messageSize;
+            _protocol = protocol;
+            _parallelCount = parallelCount;
+            _writer = resultWriter;
+
+            if (_messageCount < _parallelCount) throw new ArgumentException($"{nameof(messageCount)} needs to be greater than {nameof(parallelCount)}");
+        }
+
+        public void CreateDevice()
+        {
+            string connectionString =
+                Environment.ExpandEnvironmentVariables(Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONN_STRING"));
+
+            _client = DeviceClient.CreateFromConnectionString(connectionString, _protocol);
+
+            _wall.Start();
+        }
+
+        public async Task SendMessageAsync()
+        {
+            var metrics = new DeviceTelemetryMetrics();
+            metrics.MessageSize = _messageSize;
+            metrics.Protocol = _protocol.ToString();
+            metrics.WallTime = _wall.Elapsed.TotalMilliseconds;
+
+            try
+            {
+                Stopwatch watch = new Stopwatch();
+                watch.Restart();
+                Message message = new Message(new byte[_messageSize]);
+                Task sendTask = _client.SendEventAsync(message);
+                watch.Stop();
+                metrics.SendMs = watch.Elapsed.TotalMilliseconds;
+
+                watch.Restart();
+                await sendTask.ConfigureAwait(false);
+                watch.Stop();
+                metrics.AckMs = watch.Elapsed.TotalMilliseconds;
+            }
+            catch (Exception ex)
+            {
+                metrics.ErrorMessage = ex.Message;
+            }
+
+            await _writer.WriteAsync(metrics).ConfigureAwait(false);
+        }
+
+        public async Task RunTestAsync()
+        {
+            CreateDevice();
+
+            var tasks = new List<Task>();
+            int remainingMessages = _messageCount;
+
+
+            var sw = new Stopwatch();
+            sw.Start();
+            double startTimeSeconds = _wall.Elapsed.TotalSeconds;
+            int interimStatCompleted = 0;
+
+            for (int i = 0; i < _parallelCount; i++)
+            {
+                tasks.Add(SendMessageAsync());
```

Comment:
In the existing stress test metrics, it took about 150 seconds to initialize 50K tasks. Start `_parallelCount` tasks here is not test `DeviceClient` performance but more like dotnet task scheduler.

Reply:
I don't understand. Please clarify.
`_parallelCount` is one of the meta-parameters that need tuning based on the machine specification. In general it's small. I don't think we actually need 1000 here (I think ~100 will get the same result).